Software/vortex_ws/src/sensors/src/dvl_serial_parser.cpp

```diff
@@ -0,0 +1,80 @@
+// Copyright 2021 VorteX-co
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <algorithm>
+#include <iostream>
+#include <string>
+#include <vector>
+#include "../include/dvl_companion/parser.hpp"
+
+Parser::Parser(const std::string & packet)
+: buffer(
+    packet)               // init with input string reading, Parser p("wrx,131,..")
+{
+  setData();         // extracting measurements and setting the output variables
+}
+
+void Parser::setData()
```

Comment:
actually this as a very basic method for extracting data from the buffer , more tests on the sentences and validity checks must be added